src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/ClientCreds/JsonWebToken.cs

```diff
@@ -249,7 +247,7 @@ public string X509CertificateThumbprint
                 set { _thumbPrint = value; }
             }
 
-            [DataMember(Name = JsonWebTokenConstants.ReservedHeaderParameters.X509CertificatePublicCertValue)]
+            [DataMember(Name = JsonWebTokenConstants.ReservedHeaderParameters.X509CertificatePublicCertValue, EmitDefaultValue = false)]
```

Comment:
Found an even better way to hide the x5c claim using EmitDefaultValue set to false to ignore null properties during serialization.